pandas-stubs/_typing.pyi

```diff
@@ -53,36 +53,126 @@ from pandas.core.dtypes.dtypes import (
-Incomplete: TypeAlias = Any
+P = ParamSpec("P")
+
+HashableT = TypeVar("HashableT", bound=Hashable)
+HashableT1 = TypeVar("HashableT1", bound=Hashable)
+HashableT2 = TypeVar("HashableT2", bound=Hashable)
+HashableT3 = TypeVar("HashableT3", bound=Hashable)
+HashableT4 = TypeVar("HashableT4", bound=Hashable)
+HashableT5 = TypeVar("HashableT5", bound=Hashable)
+
+# array-like
 
 ArrayLike: TypeAlias = ExtensionArray | np.ndarray
 AnyArrayLike: TypeAlias = ArrayLike | Index[Any] | Series[Any]
+
+# list-like
+
+_T_co = TypeVar("_T_co", covariant=True)
+
+class SequenceNotStr(Protocol[_T_co]):
+    @overload
+    def __getitem__(self, index: SupportsIndex, /) -> _T_co: ...
+    @overload
+    def __getitem__(self, index: slice, /) -> Sequence[_T_co]: ...
+    def __contains__(self, value: object, /) -> bool: ...
+    def __len__(self) -> int: ...
+    def __iter__(self) -> Iterator[_T_co]: ...
+    def index(self, value: Any, start: int = ..., stop: int = ..., /) -> int: ...
+    def count(self, value: Any, /) -> int: ...
+    def __reversed__(self) -> Iterator[_T_co]: ...
+
+ListLike: TypeAlias = AnyArrayLike | SequenceNotStr | range
+
+# scalars
+
 PythonScalar: TypeAlias = str | bool | complex
-DatetimeLikeScalar = TypeVar("DatetimeLikeScalar", bound=Period | Timestamp | Timedelta)
-PandasScalar: TypeAlias = bytes | datetime.date | datetime.datetime | datetime.timedelta
+DatetimeLikeScalar: TypeAlias = Period | Timestamp | Timedelta
+PandasScalar: TypeAlias = Period | Timestamp | Timedelta | Interval
+
+_IndexIterScalar: TypeAlias = (
+    str
+    | bytes
+    | datetime.date
+    | datetime.datetime
+    | datetime.timedelta
+    | np.datetime64
+    | np.timedelta64
+    | bool
+    | int
+    | float
+    | Timestamp
+    | Timedelta
+)
+# This is wider than what is in pandas
+Scalar: TypeAlias = (
+    _IndexIterScalar | complex | np.integer | np.floating | np.complexfloating
+)
 IntStrT = TypeVar("IntStrT", int, str)
-# Scalar: TypeAlias = PythonScalar | PandasScalar
 
-DatetimeLike: TypeAlias = datetime.datetime | np.datetime64 | Timestamp
-DateAndDatetimeLike: TypeAlias = datetime.date | DatetimeLike
+# timestamp and timedelta convertible types
 
-DatetimeDictArg: TypeAlias = (
-    Sequence[int] | Sequence[float] | list[str] | tuple[Scalar, ...] | AnyArrayLike
+TimestampConvertibleTypes: TypeAlias = (
+    Timestamp
+    | datetime.datetime
+    | datetime.date
+    | np.datetime64
+    | np.int64
+    | float
+    | str
+)
+TimestampNonexistent: TypeAlias = (
+    Literal["shift_forward", "shift_backward", "NaT", "raise"]
+    | Timedelta
+    | datetime.timedelta
+)
+TimedeltaConvertibleTypes: TypeAlias = (
+    Timedelta | datetime.timedelta | np.timedelta64 | np.int64 | float | str
 )
-DictConvertible: TypeAlias = FulldatetimeDict | DataFrame
 
-CorrelationMethod: TypeAlias = (
-    Literal["pearson", "kendall", "spearman"]
-    | Callable[[np.ndarray, np.ndarray], float]
+Timezone: TypeAlias = str | tzinfo  # Not used
```

Comment:
Does "Not used" mean, it is declared in pandas but we don't use it in pandas-stubs?

Reply:
I clarified the comment.  It is declared in pandas, but not used.